ProjectManagement.BusinessLogic/Services/Implementation/FileService.cs

```diff
@@ -0,0 +1,99 @@
+        private readonly IConfiguration _configuration;
+   
+        public FileService(ILogger<FileService> logger, IConfiguration configuration)
+        {
+            _logger = logger;
+            _configuration = configuration;
+        }
+        
+        public async Task ScanFileForVirusesAsync(IFormFile file)
+        {
+            bool? isFileInfected = await IsFileInfectedAsync(file);
+            if(isFileInfected==null)
+            {
+                string message = $"The file may be infected with a virus.";
+                this._logger.LogError(message);
+                throw new WebAppException((int)HttpStatusCode.BadGateway, message);
+            }
+            else if (isFileInfected == true)
+            {
+                string message = $"The file infected with a virus.";
+                this._logger.LogError(message);
+                throw new WebAppException((int)HttpStatusCode.BadGateway, message);
+            }
+        }
+
+        public async Task<bool?> IsFileInfectedAsync(IFormFile file)
+        {
+            var ms = new MemoryStream();
+            file.OpenReadStream().CopyTo(ms);
+            byte[] fileBytes = ms.ToArray();
+            try
+            {
+                this._logger.LogInformation("ClamAV scan begin for file {0}", file.FileName);
+                var clam = new ClamClient(this._configuration["ClamAVServer:URL"],
+                                          Convert.ToInt32(this._configuration["ClamAVServer:Port"]));
+                var scanResult = await clam.SendAndScanFileAsync(fileBytes);
+                switch (scanResult.Result)
+                {
+                    case ClamScanResults.Clean:
+                        this._logger.LogInformation("The file is clean! ScanResult:{1}", scanResult.RawResult);
+                        return false;
+                    case ClamScanResults.VirusDetected:
+                        this._logger.LogError("Virus Found! Virus name: {1}", scanResult.InfectedFiles.FirstOrDefault().VirusName);
+                        return true;
+                    case ClamScanResults.Error:
+                        this._logger.LogError("An error occured while scaning the file! ScanResult: {1}", scanResult.RawResult);
+                        break;
+                    case ClamScanResults.Unknown:
+                        this._logger.LogError("Unknown scan result while scaning the file! ScanResult: {0}", scanResult.RawResult);
+                        break;
+                }
+            }
+            catch (Exception ex)
+            {
+                string message = $"ClamAV Scan Exception: {ex.ToString()}";
+                this._logger.LogError("ClamAV Scan Exception: {0}", ex.ToString());
+                throw new WebAppException((int)HttpStatusCode.BadGateway, message);
+            }
+            this._logger.LogInformation("ClamAV scan completed for file {0}", file.FileName);
+            return null;
```

Comment:
Can we return false instead of null?

Reply:
Yes we can. But null most accurately reflects the situation, when the state undefined.